clients/lsp-yaml.el

```diff
@@ -25,6 +25,11 @@
 ;;; Code:
 
 (require 'lsp-mode)
+(require 'seq)
+
+(declare-function assoc-delete-all "subr" (key alist &optional test))
+(declare-function seq-contains-p"seq" (sequence elt &optional testfn))
+(declare-function seq-filter "seq" (pred sequence))
```

Comment:
this shouldn't be needed.

Reply:
I was getting some errors from the byte compiler in the CI without these.

Reply:
This means that these function versions are not present in the seq version we depend on. We should either try to bump seq or replace them with dash versions

Reply:
Oh, and another thing I found doing some manual tests myself was that `seq-contains-p` is new in emacs 27, so I've added one more commit to avoid using it. I guess `seq-contains-p` was deprecated, so I thought I'd try to avoid any deprecation warnings in emacs 26 and use `member`. You have to covert the vector to a list to do that, which feels kind of awkward to me so I'm open suggestions.

Reply:
Regarding `seq-filter`, I've updated it to `-filter` from dash. I'm not really sure about `assoc-delete-all`- it looks like it got an optional `test` argment in emacs 27, but that isn't used here.

Reply:
In case it helps, here is a link to a failing CI job that removes the declare-function for `assoc-delete-all`: https://github.com/dellison/lsp-mode/actions/runs/386015329. I'm not sure why that happens.

Reply:
Updated this to use `assq-delete-all` instead of `assoc-delete-all`. Apparently `assoc-delete-all` has a funny history, which probably explains why this was so confusing to me.
- In Emacs 21.1, [assq-delete-all replaces function assoc-delete-all](https://github.com/emacs-mirror/emacs/blob/d0e2a341dd9a9a365fd311748df024ecb25b70ec/etc/NEWS.21#L2363)
- Later, Emacs 26.2 [introduces the new function assoc-delete-all](https://github.com/emacs-mirror/emacs/blob/master/etc/NEWS.26#L150)

My own emacs was telling me that it was "Probably introduced at or before Emacs version 21.1." in the docstring for `assoc-delete-all`. Pretty funny.